test/model/sequence_result_test.rb

```diff
@@ -3,11 +3,310 @@
+    assert_equal 0, result.error_count
+    assert_equal 0, result.todo_count
+    assert_equal 0, result.skip_count
+    assert_equal 0, result.optional_passed
+    assert_equal 0, result.optional_total
+    assert_equal 0, result.required_omitted
+    assert_equal 0, result.optional_omitted
+  end
+
+  def test_result_count
+    result = Inferno::Models::SequenceResult.new
+    result.test_results << Inferno::Models::TestResult.new
+    result.test_results << Inferno::Models::TestResult.new
+
+    assert_equal 2, result.result_count
+  end
+
+  def test_total_omitted
+    result = Inferno::Models::SequenceResult.new
+
+    result.required_omitted = 4
+    result.optional_omitted = 300
+
+    assert_equal 304, result.total_omitted
+  end
+
+  def test_update_result_counts
+    sequence_result = Inferno::Models::SequenceResult.new
+
+    test_result1 = Inferno::Models::TestResult.new
+    test_result1.required = true
+    test_result1.result = Inferno::ResultStatuses::PASS
+    sequence_result.test_results << test_result1
+
+    test_result2 = Inferno::Models::TestResult.new
+    test_result2.required = false
+    test_result2.result = Inferno::ResultStatuses::PASS
+    sequence_result.test_results << test_result2
+
+    test_result3 = Inferno::Models::TestResult.new
+    test_result3.required = true
+    test_result3.result = Inferno::ResultStatuses::OMIT
+    sequence_result.test_results << test_result3
+
+    test_result4 = Inferno::Models::TestResult.new
+    test_result4.required = false
+    test_result4.result = Inferno::ResultStatuses::OMIT
+    sequence_result.test_results << test_result4
+
+    test_result5 = Inferno::Models::TestResult.new
+    test_result5.required = true
+    test_result5.result = Inferno::ResultStatuses::TODO
+    sequence_result.test_results << test_result5
+
+    test_result6 = Inferno::Models::TestResult.new
+    test_result6.required = true
+    test_result6.result = Inferno::ResultStatuses::ERROR
+    sequence_result.test_results << test_result6
+
+    test_result7 = Inferno::Models::TestResult.new
+    test_result7.required = false
+    test_result7.result = Inferno::ResultStatuses::SKIP
+    sequence_result.test_results << test_result7
+
+    test_result8 = Inferno::Models::TestResult.new
+    test_result8.required = false
+    test_result8.result = Inferno::ResultStatuses::WAIT
+    sequence_result.test_results << test_result8
+
+    sequence_result.update_result_counts
+
+    assert_equal 1, sequence_result.required_passed
+    assert_equal 4, sequence_result.required_total
+    assert_equal 1, sequence_result.error_count
+    assert_equal 1, sequence_result.todo_count
+    assert_equal 1, sequence_result.skip_count
+    assert_equal 1, sequence_result.optional_passed
+    assert_equal 4, sequence_result.optional_total
+    assert_equal 1, sequence_result.required_omitted
+    assert_equal 1, sequence_result.optional_omitted
+  end
+
+  def test_sequence_result_result_pass
```

Comment:
Can you synchronize the function names. For example, this one is test_sequence_result_result_pass. Another one is test_sequence_result_fail.

Reply:
renamed to test_sequence_result_pass (and renamed all functions with "result_result")